### PR TITLE
chore: add construct dependencies verification to pr workflow

### PR DESCRIPTION
**Pull request by Siqi-Shan**
#### Description of changes

Add construct dependency verification to PR workflow, instead of only running script in husky commit hooks. With the change dependency verification will be automatic for all PRs.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by palpatim — APPROVED** [review]
Approved, assuming there are no node version dependencies we need to worry about

**Review by dpilch — APPROVED** [review]

**Comment by dpilch** [comment]
Noting that we can't merge this until https://github.com/aws-amplify/amplify-category-api/pull/3065 is merged.

**Comment by Siqi-Shan** [comment]
Merge this PR as dependency verification scripts improv #3078  and dependencies fix #3065  got merged.

**Pull request merged**
